spec/rubocop/cop/lint/circular_argument_reference_spec.rb

```diff
@@ -0,0 +1,71 @@
+# encoding: utf-8
+
+require 'spec_helper'
+
+describe RuboCop::Cop::Lint::CircularArgumentReference do
+  subject(:cop) { described_class.new }
+
+  before do
+    inspect_source(cop, source)
+  end
+
+  context 'when the keyword argument is not circular' do
+    let(:source) do
+      [
+        'def some_method(some_arg: nil)',
+        '  puts some_arg',
+        'end'
+      ]
+    end
+
+    it 'is happy' do
+      expect(cop.offenses.size).to eq(0)
+    end
+  end
+
+  context 'when the keyword argument is not circular, and calls a method' do
+    let(:source) do
+      [
+        'def some_method(some_arg: some_method)',
+        '  puts some_arg',
+        'end'
+      ]
+    end
+    it 'is happy' do
+      expect(cop.offenses.size).to eq(0)
+    end
+  end
+
+  context 'when there is one circular argument reference' do
+    let(:source) do
+      [
+        'def some_method(some_arg: some_arg)',
+        '  puts some_arg',
+        'end'
+      ]
+    end
+    it 'is unhappy' do
+      expect(cop.offenses.size).to eq(1)
+      expect(cop.offenses.first.message)
```

Comment:
You can also check the highlights (this is what produces the `^^^^^` under the place of the problem).